lib/wab/client.rb

```diff
@@ -0,0 +1,146 @@
+    # check against an existing object with the same key/value pairs.
+    #
+    # On error an Exception will be raised.
+    #
+    # data:: the data to use as a new object.
+    # kind:: the kind of the data. If nil the kind is taken from the data
+    # query:: query parameters to match against existing instances. A match fails the insert.
+    def create(data, kind=nil, query=nil)
+      kind = data[@type_key] if kind.nil?
+      send_request('PUT', kind, query, data)
+    end
+
+    # Return the objects according to the kind and query arguments. The
+    # following patterns supported:
+    #
+    # * query is a ref [12345] looks for MyType with reference ID of 12345
+    # * query is a Hash {name:fred,age:63} looks for all MyTypes with a name
+    #                                      of 'fred' and an age of 63.
+    #
+    # kind:: the data type
+    # query:: query parameters as a Hash.
+    def read(kind, query=nil)
+      send_request('GET', kind, query, nil)
+    end
+    
+    # Replaces the object data for the identified object. The query can be a
+    # reference to a specific object or a set of parameters to match.
+    #
+    # The return should be the identifiers for the object updated.
+    #
+    # On error an Exception should be raised.
+    #
+    # kind:: the data type
+    # data:: the data to use as a new object.
+    # query:: query parameters.
+    def update(kind, data, query)
+      raise ArgError.new('data') if data.nil?
+      raise ArgError.new('query') if query.nil?
+      send_request('POST', kind, query, data)
+    end
+
+    # Deletes the data for the identified object(s). The query can be a
+    # reference to a specific object or a set of parameters to match.
+    #
+    # The return is the identifiers for the object updated.
+    #
+    # On error an Exception should be raised.
+    #
+    # kind:: the data type
+    # query:: query parameters.
+    def delete(kind, query=nil)
+      send_request('DELETE', kind, query, nil)
+    end
+
+    # Request the server evaluate a TQL query.
+    #
+    # tql:: query to evaluate.
+    def find(tql)
+      raise ArgError.new('tql') if tql.nil?
+      send_request('POST', 'tql', nil, tql)
+    end
+    
+    private
+
+    def connect
+      return unless @http.nil?
+      @http = Net::HTTP.new(@server_address, @server_port)
+    end
+
+    def form_path(kind, query)
+      return @tql_path if kind == 'tql'
+      path = "#{@path_prefix}#{kind}"
+      if query.is_a?(Hash)
+        first = true
+        query.each_pair { |k,v|
+          if first
+            path << '?'
+            first = false
+          else
+            path << '&'
+          end
+          path << "#{k}=#{v}"
+        }
+      elsif !query.nil?
+        path << '/'
+        path << query.to_s
+      end
+      path
+    end
+    
+    def send_request(method, kind, query, content)
+      connect
+      header = {'Connection' => (@keep_alive ? 'Keep-Alive' : 'Close') }
+      unless content.nil?
```

Comment:
Rubocop would've prompted you to not use `unless..else` statements.. :smiley: 
Instead it'll tell you to re-arrange the branches to `if..else`
```ruby
if content.nil?
  resp = @http.send_request(method, form_path(kind, query))
else
  # ...
end
```

Reply:
I don't think it really matters.